### PR TITLE
[Lesson 7: Using Query Parameters] Code Assignment

### PR DESCRIPTION
**Pull request by yuliahar**
1. Task: https://learn.codethedream.org/ctd-node-lesson-7-fall2023-b/

2. Coding Assignment: implement store-api with query params

Done 11.07.2023 / deadline 11.14.2023

### CODE REVIEW

File: 04-store-api/starter/controllers/products.js
Comment (leo-mutuku) [inline 04-store-api/starter/controllers/products.js:1]: Parametrized search, sorting and field limit and pagenation well implimented.

File: 04-store-api/starter/routes/products.js
Comment (leo-mutuku) [inline 04-store-api/starter/routes/products.js:1]: get All product route well defined.

File: 04-store-api/starter/models/product.js
Comment (leo-mutuku) [inline 04-store-api/starter/models/product.js:1]: Product model well defined.

#  NOTE
IN some database ENUM are denoted in uppercase, hover in a key value pair the value can be in lowercase

File: 04-store-api/starter/populate.js
Comment (leo-mutuku) [inline 04-store-api/starter/populate.js:1]: Data migration ( product.json file) well implimented

**Review by leo-mutuku — APPROVED** [review]

File: 04-store-api/starter/models/product.js
Comment (yuliahar) [reply 04-store-api/starter/models/product.js:1]: Thanks for bringing this up. I will certainly take it into account.

File: 04-store-api/starter/populate.js
Comment (yuliahar) [reply 04-store-api/starter/populate.js:1]: Thank you for review!

**Pull request merged**